opm/material/components/BrineDynamic.hpp

```diff
@@ -0,0 +1,354 @@
+// -*- mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*-
+// vi: set et ts=4 sw=4 sts=4:
+/*
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+
+  Consult the COPYING file in the top-level source directory of this
+  module for the precise wording of the license and the list of
+  copyright holders.
+*/
+/*!
+ * \file
+ *
+ * \copydoc Opm::BrineDynamic
+ */
+#ifndef OPM_BRINEDYNAMIC_HPP
+#define OPM_BRINEDYNAMIC_HPP
+
+#include <opm/material/components/Component.hpp>
+#include <opm/material/common/MathToolbox.hpp>
+
+namespace Opm {
+
+/*!
+ * \ingroup Components
+ *
+ * \brief A class for the brine fluid properties.
+ *
+ * \tparam Scalar The type used for scalar values
+ * \tparam H2O Static polymorphism: the Brine class can access all properties of the H2O class
+ */
+template <class Scalar, class H2O>
+class BrineDynamic : public Component<Scalar, BrineDynamic<Scalar, H2O> >
+{
+public:
+
+    /*!
+     * \copydoc Component::name
+     */
+    static const char* name()
+    { return "Brine"; }
+
+    /*!
+     * \copydoc H2O::gasIsIdeal
+     */
+    static bool gasIsIdeal()
+    { return H2O::gasIsIdeal(); }
+
+    /*!
+     * \copydoc H2O::gasIsCompressible
+     */
+    static bool gasIsCompressible()
+    { return H2O::gasIsCompressible(); }
+
+    /*!
+     * \copydoc H2O::liquidIsCompressible
+     */
+    static bool liquidIsCompressible()
+    { return H2O::liquidIsCompressible(); }
+
+    /*!
+     * \copydoc Component::molarMass
+     *
+     * This assumes that the salt is pure NaCl.
+     */
+    template <class Evaluation>
+    static Evaluation molarMass(const Evaluation& salinity)
+    {
+        const Scalar M1 = H2O::molarMass();
+        constexpr Scalar M2 = 58e-3; // molar mass of NaCl [kg/mol]
+        const Evaluation X2 = salinity; // mass fraction of salt in brine
```

Comment:
Here you have a different value from  Brine_CO2.hpp (`const Scalar Ms = 58.8e-3; /* molecular weight of NaCl  [kg/mol] */`) . Shouldn't be the same value for molar mass and molar weight?